docs/pages/07_service.adoc

```diff
@@ -0,0 +1,1092 @@
+    [member] ...
+}]
+----
+
+*Example:*
+
+----
+transfer PersonTransfer(Person person) exports PersonService {
+    field String email = salesPerson.email ;
+}
+
+service PersonService(Person person) {
+    function void delete() {
+        // implement this body in Java
+    }
+}
+----
+
+The complete authorization mechanism can be defined by carefully provisioning services to actors and transfer objects. Always be aware of that if a transfer object has already been received by an external user, then they are authorized to call the functions provided by the transfer object.
+
+=== Guard
+
+A guard is a boolean expression that must evaluate to true in order to continue the execution.
+
+Guards can be defined for
+
+* functions,
+* services and
+* actors.
+
+To add a guard to functions use `guard` as follows:
+
+*Example:*
+
+----
+service PersonService(Person person) {
+    function void delete() guard person.leads!size() == 0;
+}
+----
+
+The example above defines a guard, and the delete function is not executed if that person has a lead.
+
+// TODO: what is the error message?
+
+To add a guard to services use `guard` as follows:
+
+*Example:*
+
+----
+service PersonService(Person person) guard person.leads!size() == 0 {
+    function void delete() ;
+    function Lead[] addLead(Lead lead) ;
+}
+----
+
+The example above defines a guard for service, and the functions are not executed and run-time error is thrown if that person has a lead.
+
+// TODO: add error definition to guards if error is intruduced
+
+To add a guard to actors use `guard` as follows:
+
+*Example:*
+
+----
+actor SalesPersonActor(SalesPerson salesPerson)
+    realm "COMPANY"
+    identity salesPerson.email
+    guard salesPerson.leads!size() > 0;
+----
+
+The example above defines a guard for actor, and the salesperson can only invoke the functions if she has at least one lead.
+
+=== CRUD annotations
+
+Simple new, create, delete and update functions are provided automatically. We can use `@New`, `@Create`, `@Delete` and `@Update` annotations respectively to specify these functions.
+
+
+==== @New
+
+Use `@New` annotation on a function to define a function that creates a new transfer object. The function creates the new transfer object as defined by its return type. The new transfer object will not be persisted, only the transfer object is returned.
+
+*Example:*
+
+----
+@New
+function InvoiceTransfer myNew();
+----
+
+@New annotation can be used only on functions that
+
+* have return type
+* have no parameter
+* have no body
+
+
+==== @Create and @Mapped
+
+Use `@Create` annotation on a function to define a create function. The create function persists the mapped transfer object passed as its argument. The transfer object's mapped entity must not exist. It persists the transfer object by creating a new entity.
+
+The `@Create` annotation persists only the fields of the transfer object that have the `@Mapped` annotation.
```

Comment:
`@Mapped` in combination with `@New` is problematic. Defining instance mapping in the context of `@new` is physically impossible.

Reply:
TBD
default needed